docs/adr/ADR-01-STRUCTURED_DATA.md

```diff
@@ -0,0 +1,148 @@
+2) What kind of formats/structure we want to support
+
+## How to transport variant information
+
+Certain parts of Boudicca need to know about the format of the value, for example the search service for searching, or publisher for rendering a value correctly.
+
+### Possible Solutions
+
+#### Predetermine formats for keys
+
+Also called having a schema, basically means we have a pre-defined format for every key/event property.
+
+Pros:
+* We do not have to transport any information otherwise
+* Easy to implement and maintain
+
+Cons:
+* Not flexible, users cannot introduce their own properties with other formats then the default string
+
+#### Guess formats for keys
+
+We could make guesses based on the format of the key/value and/or the query operator used to search.
+
+Pros:
+* We do not have to transport any information otherwise
+* Users can introduce their own properties with formats
+
+Cons:
+* Errorprone, we cannot guarantee correct handling
+
+#### Encode the format into the value
+
+Put the format information somewhere in the value, for example as a prefix
+
+Pros:
+* Users can introduce their own properties with formats
+
+Cons:
+* Even simple strings have to have a format prefix, otherwise we cannot guarantee correct handling
+* Unclear how to make variants of the same value, since the key would be the same
+
+#### Encode the format into the key
+
+Put the format information somewhere in the key, for example as a suffix
+
+Pros:
+* Users can introduce their own properties with formats
+* Easy to have multiple variants, since the key is different
+
+Cons:
+* Makes key-handling more complex, since there are now multiple keys for one "base-key"
+
+
+### Decision
+
+We decided on putting the format and variant information into the key, as different key=value pairs seperated by ":"
+
+At this point we need to introduce some more precise terminology:
+
+TODO move or copy this glossary somewhere else?
+* Entry: One generic entry in our database, which consists of multiple "Key"->"Value" mappings, where key and value are strings.
+* Event: A specialised "Entry", which represents an event and has two mandatory keys: "name" and "startDate"
+* Key: Key value of one "Key"->"Value" mapping of one entry. Can contain multiple variant information. Format is "PropertyName"\{:"VariantInformation"}
+* PropertyName: Name of the property of an entry. Can only consist of letters, numbers and "." TODO is that ok? do we want to allow other special chars?
+* VariantInformation: Specifying what kind of variant this key is. Format is "VariantName"="VariantValue"
+
+Currently supported variants are:
+* Language: which language a value is in
+* Format: what kind of format the value is in
+
+If a certain variant makes sense for a certain property needs to be decided per property basis.
+
+#### Language Variant
+
+Has the VariantName "lang" and the value is a two-letter language abbreviation, like "en", "de", "fr", ...
```

Comment:
how would we denote the "original" language?
Or do we use some kind of indication of collected value vs enriched value?

Reply:
hmm
maybe an event already has two languages on it's source?
i think we should just annotate generated values extra, the rest should be as is
we could think of having no lang variant as the default?